core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/metrics/MetricUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.metrics;
+
+import static software.amazon.awssdk.core.interceptor.MetricExecutionAttribute.ATTEMPT_METRIC_REGISTRY;
+import static software.amazon.awssdk.core.interceptor.MetricExecutionAttribute.METRIC_REGISTRY;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.internal.http.pipeline.stages.utils.MetricUtils;
+import software.amazon.awssdk.metrics.metrics.SdkDefaultMetric;
+import software.amazon.awssdk.metrics.registry.MetricRegistry;
+
+/**
+ * Utilities used for handling shared metric tasks during the request lifecycle.
+ */
+@SdkInternalApi
+public final class MetricUtil {
```

Comment:
How is this util class different from `MetricUtils`? Can we put them into one class?

Reply:
Didn't realize MetricUtils existed when I wrote MetricUtil and forgot to go back and combine once I did realize